pkg/query-service/app/logs/parser.go

```diff
@@ -0,0 +1,277 @@
+	"contains":  "ILIKE",
+	"ncontains": "NOT ILIKE",
+}
+
+var tokenRegex, _ = regexp.Compile(`(?i)(and( )*?)?(([\w.-]+ (in|nin) \([\S ]+\))|([\w.]+ (gt|lt|gte|lte) (')?[\S]+(')?)|([\w.]+ (contains|ncontains)) (')?[\S ]+(')?)`)
+var operatorRegex, _ = regexp.Compile(`(?i)(?: )(in|nin|gt|lt|gte|lte|contains|ncontains)(?: )`)
+
+func ParseLogFilterParams(r *http.Request) (*model.LogsFilterParams, error) {
+	res := model.LogsFilterParams{
+		Limit:   30,
+		OrderBy: "timestamp",
+		Order:   "desc",
+	}
+	var err error
+	params := r.URL.Query()
+	if val, ok := params["limit"]; ok {
+		res.Limit, err = strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+	}
+	if val, ok := params["orderBy"]; ok {
+		res.OrderBy = val[0]
+	}
+	if val, ok := params["order"]; ok {
+		res.Order = val[0]
+	}
+	if val, ok := params["q"]; ok {
+		res.Query = &val[0]
+	}
+	if val, ok := params["timestampStart"]; ok {
+		ts, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		ts64 := uint64(ts)
+		res.TimestampStart = &ts64
+	}
+	if val, ok := params["timestampEnd"]; ok {
+		ts, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		ts64 := uint64(ts)
+		res.TimestampEnd = &ts64
+	}
+	if val, ok := params["idStart"]; ok {
+		res.IdStart = &val[0]
+	}
+	if val, ok := params["idEnd"]; ok {
+		res.IdEnd = &val[0]
+	}
+	return &res, nil
+}
+
+func ParseLiveTailFilterParams(r *http.Request) (*model.LogsFilterParams, error) {
+	res := model.LogsFilterParams{}
+	params := r.URL.Query()
+	if val, ok := params["q"]; ok {
+		res.Query = &val[0]
+	}
+	if val, ok := params["timestampStart"]; ok {
+		ts, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		ts64 := uint64(ts)
+		res.TimestampStart = &ts64
+	}
+	if val, ok := params["idStart"]; ok {
+		res.IdStart = &val[0]
+	}
+	return &res, nil
+}
+
+func ParseLogAggregateParams(r *http.Request) (*model.LogsAggregateParams, error) {
+	res := model.LogsAggregateParams{}
+	params := r.URL.Query()
+	if val, ok := params["timestampStart"]; ok {
+		ts, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		ts64 := uint64(ts)
+		res.TimestampStart = &ts64
+	} else {
+		return nil, fmt.Errorf("timestampStart is required")
+	}
+	if val, ok := params["timestampEnd"]; ok {
+		ts, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		ts64 := uint64(ts)
+		res.TimestampEnd = &ts64
+	} else {
+		return nil, fmt.Errorf("timestampEnd is required")
+	}
+
+	if val, ok := params["q"]; ok {
+		res.Query = &val[0]
+	}
+
+	if val, ok := params["groupBy"]; ok {
+		res.GroupBy = &val[0]
+	}
+
+	if val, ok := params["function"]; ok {
+		res.Function = &val[0]
+	}
+
+	if val, ok := params["step"]; ok {
+		step, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, err
+		}
+		res.StepSeconds = &step
+	} else {
+		return nil, fmt.Errorf("step is required")
+	}
+	return &res, nil
+}
+
+func parseLogQuery(query string) ([]string, error) {
```

Comment:
Is there a spec/doc where I can read this query language you are using here?

Reply:
We have notes here https://www.notion.so/signoz/Draft-1-c825e4e65b564e7184dbffa5b2f8db73#238bbd26f9fe42f4b412e81c198a3f89 where we have the operators and queries.

Reply:
I was looking for some formal specification where it's easier to read what's accepted and what's rejected. Look like it is a gain regex.

Reply:
Yeah, right now the doc is more based on what kind of queries are supported rather than a spec. Will create one